network/src/primary.rs

```diff
@@ -73,6 +73,15 @@ impl PrimaryNetwork {
             }
         }
     }
+
+    pub fn cleanup<'a, I>(&mut self, to_remove: I)
+    where
+        I: IntoIterator<Item = &'a Multiaddr>,
+    {
+        for address in to_remove {
+            self.clients.remove(address);
+        }
+    }
 }
```

Comment:
Could you please add tests for this behaviour?

Reply:
@huitseeker Do you think this function could be in a nice trait? It is used by 3 out of 4 of our network structs

Reply:
@asonnino The natural place to add it is the `BaseNetwork` trait.

Reply:
But not all networks can have this functions. To be precise, they all need it except `WorkerToPrimary` that only has a single client.